packages/consent/consent-wrapper-onetrust/src/lib/onetrust-api.ts

```diff
@@ -50,10 +52,19 @@ export const getOneTrustGlobal = (): OneTrustGlobal | undefined => {
   ) {
     return oneTrust
   }
+  // if "show banner" is unchecked, window.OneTrust returns {geolocationResponse: {…}} before it actually returns the OneTrust object
+  if ('geolocationResponse' in oneTrust) {
+    return undefined
+  }
```

Comment:
Sorry - I'm having a little trouble understanding this bit.

`window.OneTrust` can be different objects depending on some conditions? Does it become the OneTrust object we know only after the banner is interacted with, or after something outside our control is loaded?

The comment about polling a few lines down confused me a bit too...when does this happen?

Reply:
> window.OneTrust can be different objects depending on some conditions?
Correct. if the banner is false, it will have a moment before it fully initialized where it just has the geolocation property. Manual testing was the only thing that uncovered that. Likely undocumented.

> The comment about polling a few lines down confused me a bit too...when does this happen?

See the usage of "resolveWhen" for when we poll for the OneTrust global

Reply:
> possible that there is some intermediate state(s) we don't know about that a customer (or us) will discover

Yeah, that seems reasonable to log then.